pandas/core/indexes/datetimelike.py

```diff
@@ -651,6 +652,18 @@ def __add__(self, other):
                 raise TypeError("cannot add {typ1} and {typ2}"
                                 .format(typ1=type(self).__name__,
                                         typ2=type(other).__name__))
+            elif isinstance(other, np.ndarray):
+                if is_timedelta64_dtype(other):
+                    return self._add_delta(other)
+                elif is_integer_dtype(other):
```

Comment:
I think instead of this, we want to specifically define __add__ in PeriodIndex which handles this case and calls super otherwise.